types/trace/trace.go

```diff
@@ -14,36 +14,55 @@ import (
-	HostName              string       `json:"hostName"`
-	ContainerID           string       `json:"containerId"`
-	Container             Container    `json:"container,omitempty"`
-	Kubernetes            Kubernetes   `json:"kubernetes,omitempty"`
-	EventID               int          `json:"eventId,string"`
-	EventName             string       `json:"eventName"`
-	MatchedPoliciesKernel uint64       `json:"-"`
-	MatchedPoliciesUser   uint64       `json:"-"`
-	MatchedPolicies       []string     `json:"matchedPolicies,omitempty"`
-	ArgsNum               int          `json:"argsNum"`
-	ReturnValue           int          `json:"returnValue"`
-	Syscall               string       `json:"syscall"`
-	StackAddresses        []uint64     `json:"stackAddresses"`
-	ContextFlags          ContextFlags `json:"contextFlags"`
-	Args                  []Argument   `json:"args"` // Arguments are ordered according their appearance in the original event
-	Metadata              *Metadata    `json:"metadata,omitempty"`
+	Timestamp             int                  `json:"timestamp"`
+	ThreadStartTime       int                  `json:"threadStartTime"`
+	ProcessorID           int                  `json:"processorId"`
+	ProcessID             int                  `json:"processId"`
+	CgroupID              uint                 `json:"cgroupId"`
+	ThreadID              int                  `json:"threadId"`
+	ParentProcessID       int                  `json:"parentProcessId"`
+	HostProcessID         int                  `json:"hostProcessId"`
+	HostThreadID          int                  `json:"hostThreadId"`
+	HostParentProcessID   int                  `json:"hostParentProcessId"`
+	UserID                int                  `json:"userId"`
+	MountNS               int                  `json:"mountNamespace"`
+	PIDNS                 int                  `json:"pidNamespace"`
+	ProcessName           string               `json:"processName"`
+	HostName              string               `json:"hostName"`
+	ContainerID           string               `json:"containerId"`
+	Container             Container            `json:"container,omitempty"`
+	Kubernetes            Kubernetes           `json:"kubernetes,omitempty"`
+	EventID               int                  `json:"eventId,string"`
+	EventName             string               `json:"eventName"`
+	MatchedPoliciesKernel uint64               `json:"-"`
+	MatchedPoliciesUser   uint64               `json:"-"`
+	MatchedPolicies       []string             `json:"matchedPolicies,omitempty"`
+	ArgsNum               int                  `json:"argsNum"`
+	ReturnValue           int                  `json:"returnValue"`
+	Syscall               string               `json:"syscall"`
+	StackAddresses        []uint64             `json:"stackAddresses"`
+	ContextFlags          ContextFlags         `json:"contextFlags"`
+	ProcIdentifier        ProcUniqueIdentifier `json:"-"`    // unique identifier for the process that generated the event
```

Comment:
Will keep the process hash for the event during the entire pipeline. Any stage in the pipeline will be able to talk to process tree.

Reply:
I recall we discussed at times a pipeline wrapper for the event, so we can separate out data needed only for the pipeline from actual fields we want to emit. I assume this would belong there (maybe add a comment).

Reply:
We've discussed in a resolved thread that the TimeSeries will be accessible by pid (i'll unresolve it so you can respond there if you want).

Reply:
Each event in a signature will have pid/starttime and ppid/ppid_starttime so u are able to pick information from the proctree very fast using the hash. 

If you need to check information for other current living (or recently exited) pids we could query a map containing latest hash for that pid (and access all information for that pid). This is very easy to have. 

Or we could even keep all hashes that PID ever had (not only the latest). 

The. of course the time series information would come from the process node.